manticore/core/cpu/disasm.py

```diff
@@ -141,6 +138,18 @@ def unimplemented(self, il):
         return (il.operation == enums.LowLevelILOperation.LLIL_UNIMPL or
                 il.operation == enums.LowLevelILOperation.LLIL_UNIMPL_MEM)
 
+    def _get_current_llil_func(self, code, pc):
+        if pc in self.llil_func_cache:
```

Comment:
There's a couple of other cases, as well. First, if you're dealing with JIT code, your cache would be invalid if a page containing JIT code was freed and replaced with a new page of JIT code. Second, if a dynamic library is unloaded and another is loaded, it could potentially be placed in the same location as the old one. In both cases, the cache would then be invalid.